rust/src/enhancers/grammar.rs

```diff
@@ -22,24 +25,83 @@ use super::matchers::{FrameOffset, Matcher};
 use super::rules::Rule;
 use super::Cache;
 
-fn ident(input: &str) -> IResult<&str, &str> {
+#[derive(Debug)]
+enum ParseErrorKind {
```

Comment:
Can we move this to a different file similar to naming: `grammer_parser.rs`? Also can we add some tests, possibly snapshot tests for the errors?